File: 45.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        vector<int> arr(nums.size(), INT_MAX);
        arr[0] = 1;
        for(int i = 0; i < nums.size(); i++){
            for(int j =1; j <= nums[i] && i+j < arr.size(); j++){
                arr[i+j] = min(arr[i+j], arr[i]+1);
            }
        }
        return arr.back()-1;
    }
};

File: Python/45.py
class Solution:    
    def jump(self, nums: List[int]) -> int:
        arr = [float('inf')]*len(nums)
        arr[0] = 0
        for i in range(len(nums)):
            for j in range(1,nums[i]+1):
                if i+j < len(nums):
                    arr[i+j] = min(arr[i]+1, arr[i+j])
        return arr[-1]
